### promise

비동기를 간편하게 처리 할 수 있게 해주는 오브젝트 => promise

const promise = new Promise(resolve, reject)=>{

}

resolve 는 정상일때

reject는 에러시.

.catch를 해야 reject 에러 설정한게 정상적으로 뜸

.then .then (213=>ads(213)) 이렇게 같을경우 출일수있음
.then(ads) 이렇게

.catch는 then 의 맨 마지막에 두면 알아서 에러 핸들링 가능 하며

해당 에러에 대한 빵구를 메꾸고 싶다면 그 해당 then 바로 뒤에 catch 로 빵구를 매우면됨

프로미스는 네트워크와 통신 파일들을 불러오는 무거운 작업들을 하기때문에

사용자가 눌러야 네트워크 통신을 해야 하는 것이라면 해당 사항을 유의 해서 만들어야된다.

### .all

promise.all([함수(),함수()])
로

병렬 실행가능

### .race()

all 과 사용방법은 똑같음

다만 둘중 하나를 레이스 시킨다는 의미인듯

먼저 들어오는거 먼저출력.
